### PR TITLE
Roll out 'statefulset' and 'daemonset' deployments also

### PR DESCRIPTION
**Pull request by avag-sargsyan**
Added functionality to roll out 'statefulset' and 'daemonset' deployments also.

UPDATE: Actually it has decided to force restart pods (by deleting) instead of deployments. Implemented that solution. Also refactored `pkg/k8s/k8s.go` in favor of making `kubeData` global.

### CODE REVIEW

**Comment by avag-sargsyan** [comment]
Need to discuss - are we moving forward with restarting deployments or restarting pods?

File: internal/core/services/k8s.go [hunk 1]
```diff
@@ -140,13 +141,32 @@ func (ks *KubeClient) GetDataFromDeployment(kubeData *models.KubeData) (*models.
...
+		ks.logger.Printf("### 🔄 Restarting the daemonset %v", kubeData.TargetName)
+		_, err = kubeData.Clientset.AppsV1().DaemonSets(kubeData.Namespace).Patch(context.Background(), kubeData.TargetName, types.StrategicMergePatchType, []byte(patchData), metav1.PatchOptions{FieldManager: "kubectl-rollout"})
+	case "statefulset":
+		// Restarting the statefulsets
+		ks.logger.Printf("### 🔄 Restarting the statefulset %v", kubeData.TargetName)
+		_, err = kubeData.Clientset.AppsV1().StatefulSets(kubeData.Namespace).Patch(context.Background(), kubeData.TargetName, types.StrategicMergePatchType, []byte(patchData), metav1.PatchOptions{FieldManager: "kubectl-rollout"})
+	default:
+		ks.logger.Printf("### 🚫 Target %v, named %v is not supported!", kubeData.TargetType, kubeData.TargetName)
+		return errors.New("target is not supported")
+	}
```
Comment (siarhei-shliayonkin) [inline internal/core/services/k8s.go]: Looks like a lot of code duplication. Can we avoid such approach?

Comment (avag-sargsyan) [reply internal/core/services/k8s.go]: Added [commit](https://github.com/ScienceSoft-Inc/integrity-sum/pull/51/commits/9c99e974568571224a057cc7013d312e1853558e)

**Comment by avag-sargsyan** [comment]
> Need to discuss - are we moving forward with restarting deployments or restarting pods?

It has been decided to restart pods instead of deployments.

File: internal/core/services/app.go [hunk 2]
```diff
@@ -113,7 +113,7 @@ func (as *AppService) Check(ctx context.Context, dirPath string, deploymentData
 			return err
 		}
 
-		err = as.IKuberService.RolloutDeployment(kuberData)
+		err = as.IKuberService.RestartPod(kuberData)
```
Comment (siarhei-shliayonkin) [inline internal/core/services/app.go]: Please update a log message (deployment -> pod)

File: internal/core/services/k8s.go [hunk 3]
```diff
@@ -139,33 +135,19 @@ func (ks *KubeClient) GetDataFromDeployment(kubeData *models.KubeData) (*models.
...
+	pName := os.Getenv("POD_NAME")
+	pNamespace := os.Getenv("POD_NAMESPACE")
 
+	// Deleting pod to force a restart
+	err := kubeData.Clientset.CoreV1().Pods(pNamespace).Delete(context.Background(), pName, metav1.DeleteOptions{})
 	if err != nil {
-		ks.logger.Printf("### 👎 Warning: Failed to patch %v, restart failed: %v", kubeData.TargetType, err)
+		ks.logger.Printf("### 👎 Warning: Failed to delete pod %v, restart failed: %v", pName, err)
 		return err
+	} else {
```
Comment (siarhei-shliayonkin) [inline internal/core/services/k8s.go]: No `else` please.

**Review by siarhei-shliayonkin — APPROVED** [review]
A few minor notices to fix.

**Review by siarhei-shliayonkin — APPROVED** [review]

File: pkg/k8s/k8s.go [hunk 4]
```diff
@@ -76,14 +114,15 @@ func (ks *KubeClient) Connect() error {
...
 
 // GetDataFromK8sAPI returns data from deployment
+// TODO: seems like it's not used
 func (ks *KubeClient) GetDataFromK8sAPI() (*DataFromK8sAPI, error) {
-	kubeData, err := ks.GetKubeData()
-	if err != nil {
-		ks.logger.Errorf("can't connect to K8sAPI: %s", err)
-		return nil, err
-	}
+	//kubeData, err := ks.GetKubeData()
```
Comment (alavrovinfb) [inline pkg/k8s/k8s.go]: if this code part ain't used, please remove.

**Review by alavrovinfb — APPROVED** [review]

File: pkg/k8s/k8s.go [hunk 5]
```diff
@@ -38,6 +38,7 @@ type DeploymentData struct {
 	ReleaseName    string
 }
 
+// TODO: seems like this struct is not needed
 type DataFromK8sAPI struct {
 	KubeData       *KubeData
 	DeploymentData *DeploymentData
```
Comment (avag-sargsyan) [inline pkg/k8s/k8s.go]: @alavrovinfb maybe I should remove this struct also? It seems we aren't gonna need it.

File: pkg/k8s/k8s.go (hunk 4 above)
Comment (avag-sargsyan) [reply pkg/k8s/k8s.go]: Done.

File: pkg/k8s/k8s.go (hunk 5 above)
Comment (alavrovinfb) [reply pkg/k8s/k8s.go]: It seems, the struct isn't used anywhere and should be removed, but please double-check it.

**Pull request merged**